src/twister2/builder/builder_abstract.py

```diff
@@ -24,13 +31,88 @@ class BuildConfig:
 class BuilderAbstract(abc.ABC):
     """Base class for builders."""
 
+    def __init__(self, build_config: BuildConfig) -> None:
+        self.build_config = build_config
+
     def __repr__(self):
         return f'{self.__class__.__name__}()'
 
     @abc.abstractmethod
-    def build(self, build_config: BuildConfig) -> None:
+    def build(self) -> None:
         """
         Build Zephyr application.
+        """
+
+    def run_cmake(self, cmake_helper: bool = False) -> None:
```

Comment:
Should we have `cmake` word in the name of method? Will we use cmake for this even in WestBuilder (I assume west will have it implemented in future)?

Reply:
Cmake word fits here, since this will make any builder to run only cmake stage. West can do this with --cmake-only arg. Do you think changing the name `run_cmake` to `run_cmake_stage` will make it clearer?

Reply:
I changed `run_cmake` into `run_cmake_stage`.